book/src/programmability/witness-pattern.md

```diff
@@ -1,277 +1,117 @@
-`T` is not stored in the struct and not stored to be used later (important security property), we
-would want to use a type that has the `drop` ability - a Witness.
-
-An updated signature of the `new_coin` would then look like this:
-
-```move
-/// Requires a `T` with the `drop` ability as a proof of ownership of `T`.
-/// However, the value is not used and discarded right away.
-public fun new_coin<T: drop>(_witness: T, value: u64): Coin<T> {
-    Coin { value }
+    /// Create a new instance of `Instance<T>` with the provided T.
+    public fun new<T>(witness: T): Instance<T> {
+        Instance { t: witness }
+    }
 }
 ```
 
-And unlike the `Container<T>`, where the `T` can be any value, including an
-[object](./../storage/key-ability.md) (hence, it can be supplied _dynamically_), the `T` in the
-`new_coin` function has to be created and consumed _statically_ - by implementing a function in the
-module that owns the `T`. So, if the type `MyCoin` is defined in the `my_coin` module, only this
-module can create a new `Coin` with `T` being `MyCoin`.
+The only way to construct an `Instance<T>` is to call the `new` function with an instance of the
+type `T`. This is a basic example of the witness pattern in Move. A module providing a witness often
+has a matching implementation, like the module `book::witness_source` below:
 
 ```move
-module book::my_coin {
-    use book::simple_coin::{Self, Coin};
+module book::witness_source {
+    use book::witness::{Self, Instance};
 
-    /// The Witness type for the Coin.
-    public struct MyCoin has drop {}
+    /// A struct used as a witness.
+    public struct W {}
 
-    /// The implementation of the `new_coin` function. Witness is not storable,
-    /// thus the only way to create a `Coin<MyCoin>` is to call this function in
-    /// this module.
-    public fun new_mycoin(value: u64): Coin<MyCoin> {
-        simple_coin::new_coin(MyCoin {}, value)
+    /// Create a new instance of `Instance<W>`.
+    public fun new_instance(): Instance<W> {
+        witness::new(W {})
     }
 }
 ```
 
-## "Abstract Class"
+The instance of the struct `W` is passed into the `new` function to create an `Instance<W>`, thereby
+proving that the module `book::witness_source` owns the type `W`.
 
-The module implementing a generic type may allow implementing certain functions on top of it. So
-that some of the functions are publicly available on the generic type, while others can be
-implemented by a specific module. This way, the generic type acts similarly to an _abstract class_
-in object-oriented languages.
+## Instantiating a Generic Type
 
-The following example demonstrates a custom `RegulatedCoin` type with implementable functions. The
-actual implementation of the `mint`, `burn`, and `transfer` functions is left to the application
-module, while the `join` function is publicly available.
+Witness allows generic types to be instantiated with a concrete type. This is useful for inheriting
```

Comment:
This section seems to be trying really hard to avoid terms like "abstract class" or "inheritance" -- is that intentional? Here's the best I could come up with without that restriction:

```
Witnesses can be used to create a pattern resembling abstract classes, where one module implements base functionality for another module to delegate to or override.
```